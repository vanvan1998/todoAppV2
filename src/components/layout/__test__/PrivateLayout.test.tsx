import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { PrivateLayout } from '..';

jest.mock('src/hooks', () => {
  const originalModule = jest.requireActual('src/hooks');
  return {
    ...originalModule,
    useMediaQuery: jest.fn().mockReturnValue({
      isMobile: true
    })
  };
});

jest.mock('next/navigation', () => {
  return {
    useRouter: jest.fn().mockReturnValue({
      push: jest.fn()
    })
  };
});

describe('Test PrivateLayout component', () => {
  test('should match private its snapshot', async () => {
    const component = render(<PrivateLayout>Test Layout</PrivateLayout>);
    expect(component).toMatchSnapshot();
  });

  test('should match its snapshot', async () => {
    const component = render(<PrivateLayout isPrivate={false}>Test Layout</PrivateLayout>);
    expect(component).toMatchSnapshot();
  });
});
