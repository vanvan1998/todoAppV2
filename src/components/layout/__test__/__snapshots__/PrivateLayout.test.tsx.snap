// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PrivateLayout component should match its snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 100%; display: flex; flex-direction: column; min-height: 100vh; background-color: rgb(239, 241, 249); align-items: center;"
      >
        <div
          class="sc-lnsxGb iqsbQF"
        >
          <a
            class=""
            href="/"
          >
            <svg
              fill="#6b7185"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M261.56 101.28a8 8 0 00-11.06 0L66.4 277.15a8 8 0 00-2.47 5.79L63.9 448a32 32 0 0032 32H192a16 16 0 0016-16V328a8 8 0 018-8h80a8 8 0 018 8v136a16 16 0 0016 16h96.06a32 32 0 0032-32V282.94a8 8 0 00-2.47-5.79z"
              />
              <path
                d="M490.91 244.15l-74.8-71.56V64a16 16 0 00-16-16h-48a16 16 0 00-16 16v32l-57.92-55.38C272.77 35.14 264.71 32 256 32c-8.68 0-16.72 3.14-22.14 8.63l-212.7 203.5c-6.22 6-7 15.87-1.34 22.37A16 16 0 0043 267.56L250.5 69.28a8 8 0 0111.06 0l207.52 198.28a16 16 0 0022.59-.44c6.14-6.36 5.63-16.86-.76-22.97z"
              />
            </svg>
          </a>
          <div
            class="sc-jSSkKI cqVDKH"
          >
            <div
              class="sc-gtLWhw jPWJOW"
              style="color: rgb(107, 113, 133);"
            >
              Hi 
            </div>
            <div>
              <button
                buttontype="primary"
                class="sc-ghWlax hYhlzt"
                data-testid="account-button"
                style="background-color: transparent;"
                type="button"
              >
                <svg
                  fill="#6b7185"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88a9.947 9.947 0 0112.28 0C16.43 19.18 14.03 20 12 20z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-giBObj dJbnCU justify-content-center"
        >
          Test Layout
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 100%; display: flex; flex-direction: column; min-height: 100vh; background-color: rgb(239, 241, 249); align-items: center;"
    >
      <div
        class="sc-lnsxGb iqsbQF"
      >
        <a
          class=""
          href="/"
        >
          <svg
            fill="#6b7185"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M261.56 101.28a8 8 0 00-11.06 0L66.4 277.15a8 8 0 00-2.47 5.79L63.9 448a32 32 0 0032 32H192a16 16 0 0016-16V328a8 8 0 018-8h80a8 8 0 018 8v136a16 16 0 0016 16h96.06a32 32 0 0032-32V282.94a8 8 0 00-2.47-5.79z"
            />
            <path
              d="M490.91 244.15l-74.8-71.56V64a16 16 0 00-16-16h-48a16 16 0 00-16 16v32l-57.92-55.38C272.77 35.14 264.71 32 256 32c-8.68 0-16.72 3.14-22.14 8.63l-212.7 203.5c-6.22 6-7 15.87-1.34 22.37A16 16 0 0043 267.56L250.5 69.28a8 8 0 0111.06 0l207.52 198.28a16 16 0 0022.59-.44c6.14-6.36 5.63-16.86-.76-22.97z"
            />
          </svg>
        </a>
        <div
          class="sc-jSSkKI cqVDKH"
        >
          <div
            class="sc-gtLWhw jPWJOW"
            style="color: rgb(107, 113, 133);"
          >
            Hi 
          </div>
          <div>
            <button
              buttontype="primary"
              class="sc-ghWlax hYhlzt"
              data-testid="account-button"
              style="background-color: transparent;"
              type="button"
            >
              <svg
                fill="#6b7185"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88a9.947 9.947 0 0112.28 0C16.43 19.18 14.03 20 12 20z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-giBObj dJbnCU justify-content-center"
      >
        Test Layout
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test PrivateLayout component should match private its snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="width: 100%; display: flex; flex-direction: column; min-height: 100vh; background-color: rgb(239, 241, 249); align-items: center;"
      >
        <div
          class="sc-lnsxGb iqsbQF"
        >
          <a
            class=""
            href="/"
          >
            <svg
              fill="#6b7185"
              height="30"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M261.56 101.28a8 8 0 00-11.06 0L66.4 277.15a8 8 0 00-2.47 5.79L63.9 448a32 32 0 0032 32H192a16 16 0 0016-16V328a8 8 0 018-8h80a8 8 0 018 8v136a16 16 0 0016 16h96.06a32 32 0 0032-32V282.94a8 8 0 00-2.47-5.79z"
              />
              <path
                d="M490.91 244.15l-74.8-71.56V64a16 16 0 00-16-16h-48a16 16 0 00-16 16v32l-57.92-55.38C272.77 35.14 264.71 32 256 32c-8.68 0-16.72 3.14-22.14 8.63l-212.7 203.5c-6.22 6-7 15.87-1.34 22.37A16 16 0 0043 267.56L250.5 69.28a8 8 0 0111.06 0l207.52 198.28a16 16 0 0022.59-.44c6.14-6.36 5.63-16.86-.76-22.97z"
              />
            </svg>
          </a>
          <div
            class="sc-jSSkKI cqVDKH"
          >
            <div
              class="sc-gtLWhw jPWJOW"
              style="color: rgb(107, 113, 133);"
            >
              Hi 
            </div>
            <div>
              <button
                buttontype="primary"
                class="sc-ghWlax hYhlzt"
                data-testid="account-button"
                style="background-color: transparent;"
                type="button"
              >
                <svg
                  fill="#6b7185"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88a9.947 9.947 0 0112.28 0C16.43 19.18 14.03 20 12 20z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="sc-giBObj dJbnCU justify-content-center"
        >
          Test Layout
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="width: 100%; display: flex; flex-direction: column; min-height: 100vh; background-color: rgb(239, 241, 249); align-items: center;"
    >
      <div
        class="sc-lnsxGb iqsbQF"
      >
        <a
          class=""
          href="/"
        >
          <svg
            fill="#6b7185"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M261.56 101.28a8 8 0 00-11.06 0L66.4 277.15a8 8 0 00-2.47 5.79L63.9 448a32 32 0 0032 32H192a16 16 0 0016-16V328a8 8 0 018-8h80a8 8 0 018 8v136a16 16 0 0016 16h96.06a32 32 0 0032-32V282.94a8 8 0 00-2.47-5.79z"
            />
            <path
              d="M490.91 244.15l-74.8-71.56V64a16 16 0 00-16-16h-48a16 16 0 00-16 16v32l-57.92-55.38C272.77 35.14 264.71 32 256 32c-8.68 0-16.72 3.14-22.14 8.63l-212.7 203.5c-6.22 6-7 15.87-1.34 22.37A16 16 0 0043 267.56L250.5 69.28a8 8 0 0111.06 0l207.52 198.28a16 16 0 0022.59-.44c6.14-6.36 5.63-16.86-.76-22.97z"
            />
          </svg>
        </a>
        <div
          class="sc-jSSkKI cqVDKH"
        >
          <div
            class="sc-gtLWhw jPWJOW"
            style="color: rgb(107, 113, 133);"
          >
            Hi 
          </div>
          <div>
            <button
              buttontype="primary"
              class="sc-ghWlax hYhlzt"
              data-testid="account-button"
              style="background-color: transparent;"
              type="button"
            >
              <svg
                fill="#6b7185"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 4c1.93 0 3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5S10.07 6 12 6zm0 14c-2.03 0-4.43-.82-6.14-2.88a9.947 9.947 0 0112.28 0C16.43 19.18 14.03 20 12 20z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
      <div
        class="sc-giBObj dJbnCU justify-content-center"
      >
        Test Layout
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
